model_compression_toolkit/gptq/keras/gptq_training.py

```diff
@@ -77,6 +81,15 @@ def __init__(self,
             hessian_info_service: HessianScoresService for fetching and computing Hessian's approximation scores.
 
         """
+
+        def _get_total_grad_steps():
+            return get_total_grad_steps(representative_data_gen) * gptq_config.n_epochs
+
+        # must be set prior to model building in the base class constructor. TODO: why?
```

Comment:
why really?

Reply:
it is used in activation holder construction so it must be set before the model is built, which happens in the base constructor